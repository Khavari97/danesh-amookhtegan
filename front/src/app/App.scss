@import 'assets/fonts/fonts';

* {
  margin: 0;
  font-family: "Shabnam", serif;
}

a {
  text-decoration: none;
}

//Slider css
.alice-carousel__dots {
  .alice-carousel__dots-item:not(.__custom).__active {
    background-color: #046B00;
  }
}


