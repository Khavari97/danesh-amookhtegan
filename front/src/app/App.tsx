import React from "react";
import "./App.scss";
import AppRouter from "./navigation/AppRouter";
import {SnackbarProvider} from 'notistack';
import {SnackbarUtils} from "./components/material-ui/Snackbar/SnackbarUtils";
import ThemeProvider from "./assets/styles/material/material";

export interface AppProps {
}

export interface AppState {
}

class App extends React.Component<AppProps, AppState> {
    constructor(props: AppProps) {
        super(props);
        this.state = {};
    }

    componentDidMount = () => {
    }

    render() {
        return (
            <SnackbarProvider maxSnack={3} anchorOrigin={{horizontal: 'right', vertical: 'bottom'}}>
                <SnackbarUtils/>
                <ThemeProvider>
                    <AppRouter/>
                </ThemeProvider>
            </SnackbarProvider>
        );
    }
}

export default App;
