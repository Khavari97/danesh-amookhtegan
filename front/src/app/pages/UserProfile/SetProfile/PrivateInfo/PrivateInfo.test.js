import React from 'react';
import Enzyme, { shallow, configure, mount } from "enzyme";
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
import { Provider } from "react-redux";
import App from "./PrivateInfo";
import renderer from "react-test-renderer";
import { render, screen, act } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom'
import { Button } from '@material-ui/core';
import thunk from 'redux-thunk'
import configureStore from 'redux-mock-store';
configure({ adapter: new Adapter() })

const middlewares = [thunk]
const mockStore = configureStore(middlewares);
const findByTestAttr = (wrapper, val) => {
    return wrapper.find(`[data-testid="${val}"]`);
};



describe('test set profile states', () => {
    let wrapper
    let store_mock
    const data = {
        city: "تهران",
        province: "تهران",
        email: "atena.ganji.ag@gmail.com",
        firstName: "آتنا",
        gender: "FEMALE",
        genderDisplay: "زن",
        lastName: "گنجی",
        phoneNumber: "+989011625458",
        university: "خوارزمی",
        username: "atena",
    }

    store_mock = mockStore({ userProfile: { loading: false, data: data } })
    store_mock.dispatch = jest.fn()

    wrapper = mount(
        <BrowserRouter>
            <Provider store={store_mock}>
                <App />
            </Provider>
        </BrowserRouter>
    );


    test('check show props', () => {
        expect(findByTestAttr(wrapper, 'username').text()).toEqual(data.username)
        expect(findByTestAttr(wrapper, 'phoneNumber').text()).toEqual(data.phoneNumber)
        expect(findByTestAttr(wrapper, 'university').text()).toEqual(data.university)
        expect(findByTestAttr(wrapper, 'city').text()).toEqual(data.city)
        expect(findByTestAttr(wrapper, 'province').text()).toEqual(data.province)
        expect(findByTestAttr(wrapper, 'email').text()).toEqual(data.email)
        expect(findByTestAttr(wrapper, 'province').text()).toEqual(data.province)
        expect(findByTestAttr(wrapper, 'genderDisplay').text()).toEqual(data.genderDisplay)
    })

    test('snapshot', () => {
        const snapshot = renderer.create(wrapper).toJSON();
        expect(snapshot).toMatchSnapshot();
    })

});