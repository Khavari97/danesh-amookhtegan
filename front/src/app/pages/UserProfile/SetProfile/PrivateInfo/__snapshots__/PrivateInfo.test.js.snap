// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test set profile states snapshot 1`] = `
<div
  className="MuiGrid-root privateInfo animate__animated animate__fadeIn MuiGrid-container"
>
  <div
    className="MuiGrid-root gridItems MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <div>
      <li
        className="MuiListItem-root MuiListItem-gutters"
        disabled={false}
      >
        <p
          className="title"
        >
          نام
        </p>
      </li>
      <li
        className="MuiListItem-root MuiListItem-gutters"
        disabled={false}
      >
        <p
          className="description"
          data-testid="firstName"
        >
          آتنا
        </p>
      </li>
    </div>
    <div>
      <li
        className="MuiListItem-root MuiListItem-gutters"
        disabled={false}
      >
        <p
          className="title"
        >
          نام خانوادگی
        </p>
      </li>
      <li
        className="MuiListItem-root MuiListItem-gutters"
        disabled={false}
      >
        <p
          className="description"
          data-testid="lastName"
        >
          گنجی
        </p>
      </li>
    </div>
  </div>
  <div
    className="MuiGrid-root gridItems MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <div>
      <li
        className="MuiListItem-root MuiListItem-gutters"
        disabled={false}
      >
        <p
          className="title"
        >
          جنسیت
        </p>
      </li>
      <li
        className="MuiListItem-root MuiListItem-gutters"
        disabled={false}
      >
        <p
          className="description"
          data-testid="genderDisplay"
        >
          زن
        </p>
      </li>
    </div>
  </div>
  <div
    className="MuiGrid-root gridItems MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <div>
      <li
        className="MuiListItem-root MuiListItem-gutters"
        disabled={false}
      >
        <p
          className="title"
        >
          نام کاربری
        </p>
      </li>
      <li
        className="MuiListItem-root MuiListItem-gutters"
        disabled={false}
      >
        <p
          className="description"
          data-testid="username"
        >
          atena
        </p>
      </li>
    </div>
  </div>
  <div
    className="MuiGrid-root gridItems MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <div>
      <li
        className="MuiListItem-root MuiListItem-gutters"
        disabled={false}
      >
        <p
          className="title"
        >
          ایمیل
        </p>
      </li>
      <li
        className="MuiListItem-root MuiListItem-gutters"
        disabled={false}
      >
        <p
          className="description"
          data-testid="email"
        >
          atena.ganji.ag@gmail.com
        </p>
      </li>
    </div>
  </div>
  <div
    className="MuiGrid-root gridItems MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <div>
      <li
        className="MuiListItem-root MuiListItem-gutters"
        disabled={false}
      >
        <p
          className="title"
        >
          استان
        </p>
      </li>
      <li
        className="MuiListItem-root MuiListItem-gutters"
        disabled={false}
      >
        <p
          className="description"
          data-testid="province"
        >
          تهران
        </p>
      </li>
    </div>
    <div>
      <li
        className="MuiListItem-root MuiListItem-gutters"
        disabled={false}
      >
        <p
          className="title"
        >
          شهر
        </p>
      </li>
      <li
        className="MuiListItem-root MuiListItem-gutters"
        disabled={false}
      >
        <p
          className="description"
          data-testid="city"
        >
          تهران
        </p>
      </li>
    </div>
  </div>
  <div
    className="MuiGrid-root gridItems MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <div>
      <li
        className="MuiListItem-root MuiListItem-gutters"
        disabled={false}
      >
        <p
          className="title"
        >
          دانشگاه
        </p>
      </li>
      <li
        className="MuiListItem-root MuiListItem-gutters"
        disabled={false}
      >
        <p
          className="description"
          data-testid="university"
        >
          خوارزمی
        </p>
      </li>
    </div>
  </div>
  <div
    className="MuiGrid-root gridItems MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6"
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <div>
      <li
        className="MuiListItem-root MuiListItem-gutters"
        disabled={false}
      >
        <p
          className="title"
        >
          شماره تلفن
        </p>
      </li>
      <li
        className="MuiListItem-root MuiListItem-gutters"
        disabled={false}
      >
        <p
          className="description"
          data-testid="phoneNumber"
        >
          +989011625458
        </p>
      </li>
    </div>
  </div>
</div>
`;
